addon/modifiers/did-resize.js

```diff
@@ -6,23 +6,46 @@ export default class DidResizeModifier extends Modifier {
   options = {};
 
   // Private API
-  observer = null;
+  static observer = null;
+  static handlers = null;
 
-  observe() {
-    if (this.observer) {
-      this.observer.observe(this.element, this.options);
+  constructor() {
+    super(...arguments);
+
+    if (!('ResizeObserver' in window)) {
+      return;
+    }
+
+    if (!DidResizeModifier.observer) {
+      DidResizeModifier.handlers = new WeakMap();
+      DidResizeModifier.observer = new ResizeObserver((entries, observer) => {
+        for (let entry of entries) {
+          const handler = DidResizeModifier.handlers.get(entry.target);
```

Comment:
I am confident this works as expected since this method will only get set up once, but since we're sharing this across all instances, is it possible to write a test that asserts each handler is only called once per observer event, when there are multiple modifier instances rendered? I just want to make sure that we have a test in place in case someone else comes along and changes this implementation in the future, since they might not immediately understand how this works.

Reply:
Good idea! I've added the new test case.